[hipblaslt] Support BSS GEMM problems on gfx90a

chuckwu20000 opened the pull request:
## Motivation

<!-- Explain the purpose of this PR and the goals it aims to achieve. -->
According to the SWDEV-560157 & SWDEV-561151, pytorch team recently enable overload gemm tests where the inputs are bf16 and the output is fp32. The solution library for gfx90a currently lacks support for BSS GEMM problems.

## Technical Details

<!-- Explain the changes along with any relevant GitHub links. -->
This PR adds the logic YAML files that was tuned using Tensile.
Problems not support: with epilogues

## Test Plan

<!-- Explain any relevant testing done to verify this PR. -->
set env variable: `HIPBLASLT_LOG_MASK` to 0xFF
1. Test hipblaslt-bench with `--transA T/N --transB T/N --a_type bf16_r --b_type bf16_r --c_type f32_r --d_type f32_r --compute_type f32_r --verify`
2. Add test into matmul_gtest

## Test Result

<!-- Briefly summarize test outcomes. -->
1. Confirmed that BSS GEMM problems can now find solutions, instead of returning the error message: `NO solution found!`.
2. Pass the matmul_bf16_fp32dst gtest on gfx90a

## Submission Checklist

- [X] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

jichangjichang commented:
The solution put in 104cu folder would be specified as 104CU in logic yaml. The default solution pool of MI200 serials is 110CU.
Search this keyword for your referece:
- {Architecture: gfx90a, CUCount: 104}

Your logic yaml is not specified for 104cu, then you should put the logic yaml to non-specified CU folder (110CU) which is the fallback solution pool for all CUs.

Should we support other transpose and full epilogues for BSS as well?
Bias, Scale alpha vector, Activation

jichangjichang commented:
From logs of SWDEV-560157, the test is not only for TT but NN. I think you should enable 4 transpose.

chuckwu20000 commented:
> From logs of SWDEV-560157, the test is not only for TT but NN. I think you should enable 4 transpose.

Thank you for checking. I have added 4 transpose's logic yaml and moved them to the 110CU dir.
And this PR aims to help the framework team pass testing asap. Additional epilogue requests will be added in the future updates.

jichangjichang commented:
I think you can enable "matmul_bf16_fp32dst" Gtest for gfx90a as well.
